### PR TITLE
Set `LangVersion` compiler option to `11.0`

### PR DESCRIPTION
**Pull request by xtqqczze**

### CODE REVIEW

**Comment by iSazonov** [comment]
I think we need to switch to C# 11 explicitly (not latest).

File: test/Test.Common.props [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <Copyright>(c) Microsoft Corporation.</Copyright>
 
     <TargetFramework>net7.0</TargetFramework>
-    <LangVersion>11.0</LangVersion>
+    <LangVersion>latest</LangVersion>
```
Comment (daxian-dbw) [inline test/Test.Common.props:10]: May be keep it unchanged here?

Comment (xtqqczze) [reply test/Test.Common.props]: Sorry, bad merge. Rebased.

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Comment by daxian-dbw** [comment]
@xtqqczze The changes here will conflict with #19194. Given that this PR contains way less changes than #19194, I think it's better to merge the other one first. You will have to rebase and resolve the conflicts after the other one gets in. Thanks!

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Pull request merged**